contracts/strategies/aave/StrategyAaveLendingPool.sol

```diff
@@ -0,0 +1,230 @@
+                return 1e18;
+            }
+            return MAX_INT;
+        }
+        return (assetAmount * 1e18) / shares;
+    }
+
+    function harvest() external override onlyEOA {
+        // 1. Claims the liquidity incentives
+        address[] memory assets = new address[](1);
+        assets[0] = aToken;
+        uint256 rewardsBalance =
+            IAaveIncentivesController(incentivesController).getRewardsBalance(assets, address(this));
+        if (rewardsBalance > 0) {
+            IAaveIncentivesController(incentivesController).claimRewards(assets, rewardsBalance, address(this));
+        }
+
+        // 2. Activates the cooldown period if not already activated
+        uint256 stakedAaveBalance = IERC20(stakedAave).balanceOf(address(this));
+        if (stakedAaveBalance > 0 && IStakedAave(stakedAave).stakersCooldowns(address(this)) == 0) {
+            IStakedAave(stakedAave).cooldown();
+        }
+
+        // 3. Claims the AAVE staking rewards
+        uint256 stakingRewards = IStakedAave(stakedAave).getTotalRewardsBalance(address(this));
+        if (stakingRewards > 0) {
+            IStakedAave(stakedAave).claimRewards(address(this), stakingRewards);
+        }
+
+        // 4. Redeems staked AAVE if possible
+        uint256 cooldownStartTimestamp = IStakedAave(stakedAave).stakersCooldowns(address(this));
+        if (
+            stakedAaveBalance > 0 &&
+            block.timestamp > cooldownStartTimestamp.add(IStakedAave(stakedAave).COOLDOWN_SECONDS()) &&
+            block.timestamp <=
+            cooldownStartTimestamp.add(IStakedAave(stakedAave).COOLDOWN_SECONDS()).add(
+                IStakedAave(stakedAave).UNSTAKE_WINDOW()
+            )
+        ) {
+            IStakedAave(stakedAave).redeem(address(this), stakedAaveBalance);
+        }
+
+        // 5. Sells the reward AAVE token and the redeemed staked AAVE for obtain more supplying token
+        uint256 aaveBalance = IERC20(aave).balanceOf(address(this));
+        if (aaveBalance > 0) {
+            IERC20(aave).safeIncreaseAllowance(uniswap, aaveBalance);
+
+            address[] memory paths = new address[](3);
+            paths[0] = aave;
+            paths[1] = weth;
+            paths[2] = supplyToken;
+
+            IUniswapV2(uniswap).swapExactTokensForTokens(
+                aaveBalance,
+                uint256(0),
+                paths,
+                address(this),
+                block.timestamp.add(1800)
+            );
+
+            // Deposit supplying token to Aave Lending Pool and mint aToken.
+            uint256 obtainedSupplyTokenAmount = IERC20(supplyToken).balanceOf(address(this));
+            IERC20(supplyToken).safeIncreaseAllowance(lendingPool, obtainedSupplyTokenAmount);
+            ILendingPool(lendingPool).deposit(supplyToken, obtainedSupplyTokenAmount, address(this), 0);
+        }
+
+        // sync the asset balance
+        assetAmount = IAToken(aToken).balanceOf(address(this));
```

Comment:
I don't think we need to store and keep track of `assetAmount`. `syncPrice` should query it.

Reply:
updated